### ISSUE TITLE
Add support for `Bundle` inside `BundleItem`

### ISSUE THREAD

**Issue by ghost** [mattsse/mev-share-rs#44]
What's the reason that `mev-share-rs` doesn't support using `Bundle` as a [`BundleItem`](https://github.com/paradigmxyz/mev-share-rs/blob/93c8ef00c8c840b69e2dc8107f1e4212d7eb6dfa/crates/mev-share-rpc-api/src/types.rs#L66)? 

I see that [MEV-Share supports this](https://docs.flashbots.net/flashbots-mev-share/searchers/sending-bundles#nest-bundles-be-a-mev-share-node).

```rust
pub enum BundleItem {
    /// The hash of either a transaction or bundle we are trying to backrun.
    Hash {
        /// Tx hash.
        hash: TxHash,
    },
    /// A new signed transaction.
    #[serde(rename_all = "camelCase")]
    Tx {
        /// Bytes of the signed transaction.
        tx: Bytes,
        /// If true, the transaction can revert without the bundle being considered invalid.
        can_revert: bool,
    },
}
```

**Comment by mattsse** [comment]
yeah I think we could actually move all of these types into the alloy mev crate:

https://github.com/paradigmxyz/mev-share-rs/blob/main/crates/mev-share-sse/src/types.rs

**Comment by SkandaBhat** [comment]
`Bundle` has been added to `BundleItem` on alloy: https://github.com/alloy-rs/alloy/pull/1418 Should it be added in this repo as well? @mattsse
